### PR TITLE
add first version of py-pyrr package

### PR DESCRIPTION
**Pull request by JeromeDuboisPro**
Add py-pyrr package, useful for maths operations, rays and objects manipulations for raytracing.

### CODE REVIEW

**Comment by JeromeDuboisPro** [comment]
@tgamblin @scheibelp poking around to try to get this PR merged. Ok for you?

**Review by tldahlgren — DISMISSED** [review]
Confirmed sha256.  Deferring merge decision until @adamjstewart has a chance to comment.

File: var/spack/repos/builtin/packages/py-pyrr/package.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+class PyPyrr(PythonPackage):
+    """3D mathematical functions using NumPy."""
+
+    homepage = "https://github.com/adamlwgriffiths/Pyrr"
+    pypi     = "pyrr/pyrr-0.10.3.tar.gz"
+    maintainers = ['JeromeDuboisPro']
+    license = "BSD-2-Clause"
+
+    version('0.10.3', sha256='3c0f7b20326e71f706a610d58f2190fff73af01eef60c19cb188b186f0ec7e1d', preferred=True)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyrr/package.py]:
```suggestion
    version('0.10.3', sha256='3c0f7b20326e71f706a610d58f2190fff73af01eef60c19cb188b186f0ec7e1d')
```

File: var/spack/repos/builtin/packages/py-pyrr/package.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+    depends_on('py-setuptools', type='build')
+    depends_on('py-numpy')
+    depends_on('py-multipledispatch')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyrr/package.py]:
```suggestion
    depends_on('py-numpy', type=('build', 'run'))
    depends_on('py-multipledispatch', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-pyrr/package.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+from spack.package import *
+
+
+class PyPyrr(PythonPackage):
+    """3D mathematical functions using NumPy."""
+
+    homepage = "https://github.com/adamlwgriffiths/Pyrr"
+    pypi     = "pyrr/pyrr-0.10.3.tar.gz"
+    maintainers = ['JeromeDuboisPro']
+    license = "BSD-2-Clause"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-pyrr/package.py]: This variable isn't used by Spack as far as I know, can probably remove this.

**Review by adamjstewart — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-pyrr/package.py (hunk 3 above)
Comment (JeromeDuboisPro) [reply var/spack/repos/builtin/packages/py-pyrr/package.py]: I guess you are refering to the license variable.
During linux checks, I had a failure due to header or license being wrong.
I found a "license=" in some other py-* package, so I tried this. Turned out it was the copyright year range "-2020" -> "-2022" which fixed this failure.
I'll remove the license variable to resolve this item.
Thanks a lot for all the appropriate comments and suggestions 😊.

Comment (JeromeDuboisPro) [reply var/spack/repos/builtin/packages/py-pyrr/package.py]: removed license variable with commit 80e2f648af048cc0cfd9a40a9224336af664e0d0

**Comment by JeromeDuboisPro** [comment]
@adamjstewart should be good. Ok to you?

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**